Dev

davidsantiagoquevedo opened the pull request:
* Fixed coherence issues with the example data set
* Added checks for date format
* Updated and fixed examples
* Fixed checks in get immunizing vaccine and dose for more than one vaccine
* Fixed tests for both functions

pratikunterwegs commented:
Hey @davidsantiagoquevedo, thanks for tagging me - I'm happy to review, and I can try to get this done by Friday but it might take a bit longer. I'll be on leave all of next week so I might only be able to give feedback in the week of the 11th if that's okay.

davidsantiagoquevedo commented:
Hi @pratikunterwegs. Yes, it's okay. Thanks for your time.

pratikunterwegs commented on R/coh_data_wrangling.R:
```diff
@@ -168,27 +167,35 @@ get_immunization_date <- function(data,
-  # expect end cohort is a string - will be coerced to date later
-  checkmate::assert_string(
+  # expect end cohort is a date
+  checkmate::assert_date(
     end_cohort
   )
 
```
If I understand correctly the end of the study is given by a single date, in which case it would be good to check for a single, non-missing value.
```suggestion
  checkmate::assert_date(
    end_cohort, any.missing = FALSE, len = 1
  )

```

davidsantiagoquevedo replied:
Right, thanks for the suggestion.

pratikunterwegs commented on R/coh_data_wrangling.R:
```diff
@@ -352,9 +363,14 @@ get_time_to_event <- function(data, outcome_date_col,
-  checkmate::assert_string(start_cohort)
-  checkmate::assert_string(end_cohort)
+  # check date types
+  checkmate::assert_date(start_cohort)
+  checkmate::assert_date(end_cohort)
```
Check for single length and no missing values here.

davidsantiagoquevedo replied:
Done

pratikunterwegs commented on R/coh_data_wrangling.R:
```diff
@@ -352,9 +363,14 @@ get_time_to_event <- function(data, outcome_date_col,
+  checkmate::assert_date(start_cohort)
+  checkmate::assert_date(end_cohort)
+
+  # check if date columns are date type
+  checkmate::assert_date(
+    data[[outcome_date_col]]
+  )
```
Not sure whether you should allow missing values in this column, would be good to check with Santiago. If not, add a check here using `any.missing = FALSE`.

davidsantiagoquevedo replied:
Missing values are allowed in all the date-type columns. In the cohort study, the population is split into those with and without outcome, based on whether they have or not a date reported.

pratikunterwegs commented on R/coh_data_wrangling.R:
```diff
@@ -371,16 +387,11 @@ get_time_to_event <- function(data, outcome_date_col,
   if (start_from_immunization) {
     # convert date columns to Date type
-    data[[immunization_date_col]] <- as.Date(data[[immunization_date_col]])
+    checkmate::assert_date(
+      data[[immunization_date_col]]
+    )
```
Same as above - confirm whether missing values are allowed here.

davidsantiagoquevedo replied:
Allowed. In this case not having immunization date means unvaccinated.

pratikunterwegs commented on R/coh_data_wrangling.R:
```diff
@@ -498,25 +509,36 @@ get_immunization_dose <- function(data,
     colnames(data),
     must.include = c(vacc_date_col, immunization_date_col)
   )
+  checkmate::assert_date(
+    data[[immunization_date_col]]
+  )
+  # The function can receive one or more column names in vacc_date_col
+  vacc_date_col_ <- vacc_date_col # hard coded to return right error message
+  for (vacc_date_col in c(vacc_date_col_)) {
+    checkmate::assert_date(
+      data[[vacc_date_col]]
+    )
+  }
```
Again, confirm whether missing values are allowed in date columns.

davidsantiagoquevedo replied:
Allowed.

pratikunterwegs commented on R/coh_data_wrangling.R:
```diff
@@ -498,25 +509,36 @@ get_immunization_dose <- function(data,
+    data[[immunization_date_col]]
+  )
+  # The function can receive one or more column names in vacc_date_col
+  vacc_date_col_ <- vacc_date_col # hard coded to return right error message
```
Is this extra assignment to a copy necessary? The loop below works fine as so:

```
for(i in vacc_date_col) { assert_date(data[[i]] }
```

The extra call to `c()` in `c(vacc_date_col_)` also seems unnecessary.

davidsantiagoquevedo replied:
I think it's necessary because when the iterator is `i`  the assertion error returns:
`Error: Assertion on 'data[[i]]' failed: Must be of class 'Date', not 'character'.`

This could be a bit confusing for the users (they won't know which one is the variable `i`). I would rather the message:
`Error: Assertion on 'data[[vacc_date_col]]' failed: Must be of class 'Date', not 'character'.`

davidsantiagoquevedo replied:
The extra call to `c()` is indeed unnecessary

pratikunterwegs commented on R/coh_data_wrangling.R:
```diff
@@ -498,25 +509,36 @@ get_immunization_dose <- function(data,
   # get the first dose corresponding to immunization date - delay
-  dose_number <- apply(data[, c(vacc_date_col, "delta_imm")], 1, function(x) {
+  dose_number <- apply(data[, c(vacc_date_col_, "delta_imm")], 1, function(x) {
     which(x == x[length(x)])[1] # hard coded to get first value
   })
 
   # get names of the vaccination columns corresponding to the dose
-  return(vacc_date_col[dose_number])
+  return(vacc_date_col_[dose_number])
```
See suggestion above - these changes could be reverted.

davidsantiagoquevedo replied:
I suggest keeping them, based on the previous comment.

pratikunterwegs commented on README.md:
This file looks like it has been generated offline using `devtools::build_readme()`. Could you push a small change to `Readme.Rmd` so that the `render-readme` workflow is triggered and rebuilds this file with the package name and repo name in the right places?

davidsantiagoquevedo replied:
Yes, it was my mistake. Done

pratikunterwegs commented on tests/testthat/test-set_status.R:
```diff
@@ -30,7 +31,7 @@ test_that("`set_status`: Basic expectations", {
 
   # expect unique values
   expect_identical(
-    unique(vaccine_status),
+    sort(unique(vaccine_status), decreasing = TRUE),
     vax_status
   )
 })
```
This is probably a mistake I introduced in #13, but it can be fixed by using `expect_setequal()`. No need for `unique()` as well.

```
expect_setequal(vaccine_status, vax_status)
```

pratikunterwegs reviewed (changes requested):
Looks good overall @davidsantiagoquevedo - the checks are currently failing because of an `rds` file in the wrong place - if you remove that, I think it should be alright. Just a few other points which I've put down below as comments on the files; one is that the Readme.md needs to be rebuilt. The test coverage is already high but would be good to bring to 100%; try running `goodpractice::gp()` to identify which lines are missed, or check Codecov for this branch.

davidsantiagoquevedo commented:
Hi @pratikunterwegs, 
Thanks for the review. I addressed most of the suggestions including regenerating `cohortdata`. The `9625 marked UTF-8 strings` came from the example datasets for the screening method and test-negative. I temporarily removed both datasets because we're not using them, and they must be cleaned and processed before starting the actual implementation of both methods. They were just causing noise. I also noticed that `goodpractice::gp()` was returning a warning related to the size of the datasets, we solved this with @jpavlich using `tools::resaveRdaFiles("inst/extdata/")` and `tools::resaveRdaFiles("data/")`.

I'll continue working on the test coverage. However, all the R-CMD-checks are passing now.

pratikunterwegs reviewed (approved):
Hi @davidsantiagoquevedo thanks for fixing these - looks good to me.
